RELEASING.md

```diff
@@ -1,8 +1,72 @@
-### Overview
-
-The release process for SNAPSHOT versions is handled by the `after_success.sh` script, triggered after Travis CI successfully completes a non-PR build. A new SNAPSHOT artifact will be released to the snapshot repository as part of this script.
-
-### To push a release version
```

Comment:
Done. As I was making this consistent with instructions for google-auth-library-java and gax-java, I will make the same change in those places.